src/sentry/deletions/base.py

```diff
@@ -270,30 +269,18 @@ class BulkModelDeletionTask(ModelDeletionTask[ModelT]):
 
     DEFAULT_CHUNK_SIZE = 10000
 
-    def __init__(
-        self,
-        manager: DeletionTaskManager,
-        model: type[ModelT],
-        query: Mapping[str, Any],
-        partition_key: str | None = None,
-        **kwargs: Any,
-    ):
-        super().__init__(manager, model, query, **kwargs)
-
-        self.partition_key = partition_key
-
     def chunk(self) -> bool:
         return self._delete_instance_bulk()
 
     def _delete_instance_bulk(self) -> bool:
         try:
-            return bulk_delete_objects(
-                model=self.model,
-                limit=self.chunk_size,
-                transaction_id=self.transaction_id,
-                partition_key=self.partition_key,
-                **self.query,
-            )
+            with unguarded_write(using=router.db_for_write(self.model)):
```

Comment:
this was previously bypassing unguarded write checking because the manual sql did not match the HC regex so this is ~roughly equivalent